includes/common.php

```diff
@@ -821,10 +821,37 @@ function version_info($remote = false)
     $output['netsnmp_ver'] = str_replace('version: ', '', rtrim(shell_exec(
         Config::get('snmpget', 'snmpget') . ' -V 2>&1'
     )));
+    $output['openssl_ver'] = explode(' ', rtrim(shell_exec(
+        Config::get('openssl', 'openssl') . ' version 2>&1'
+    )))[1];
 
     return $output;
 }//end version_info()
 
+/**
+ * checks if System is SNMPv3 SHA2 Capable for Auth Algorithms (SHA-224,SHA-256,SHA-384,SHA-512)
+ * @return boolean
+ */
+function snmpv3_sha2_capable()
```

Comment:
This doesn't feel like the right way to do this.  Is there a possibility to have other SSL libraries than openssl?
Also, this is not a lightweight function and you call it many times.
Your version compare is incorrect and it will cast the strings to floats.  you need to use version_compare()

Reply:
@murrant from [here](https://fossies.org/diffs/net-snmp/5.7.3_vs_5.8/configure-diff.html) it seems that test for OpenSSL themselves? But I am not 1000% sure.

![image](https://user-images.githubusercontent.com/24249616/89175587-4fd74d00-d588-11ea-936c-91153d0809db.png)

Will fix the validation.

Reply:
$ snmpget --help 2>&1 |grep PROTOCOL
  -a PROTOCOL           set authentication protocol (MD5|SHA|SHA-224|SHA-256|SHA-384|SHA-512)
  -x PROTOCOL           set privacy protocol (DES|AES)

Let net-snmp do the checks, and that way we don't need to call openssl at all.  Just parse the output.